pkg/tsdb/azuremonitor/azuremonitor.go

```diff
@@ -188,7 +189,7 @@ func checkAzureMonitorMetricsHealth(dsInfo types.DatasourceInfo) (*http.Response
 
 	res, err := dsInfo.Services["Azure Monitor"].HTTPClient.Do(request)
 	if err != nil {
-		return nil, err
+		return nil, types.AzureError(err)
```

Comment:
if you define the error as an AzureError, you can return wrap it with `%w` so you can check its type later:

```go
			return nil, fmt.Errorf("%w: %v", types.AzureError, err)
```